public/app/features/scenes/variables/variants/DataSourceVariable.tsx

```diff
@@ -0,0 +1,109 @@
+import React from 'react';
+import { Observable, of } from 'rxjs';
+
+import { stringToJsRegex, VariableRefresh, DataSourceInstanceSettings } from '@grafana/data';
+import { getDataSourceSrv } from '@grafana/runtime';
+
+import { sceneGraph } from '../../core/sceneGraph';
+import { SceneComponentProps } from '../../core/types';
+import { VariableDependencyConfig } from '../VariableDependencyConfig';
+import { VariableValueSelect } from '../components/VariableValueSelect';
+import { VariableValueOption } from '../types';
+
+import { MultiValueVariable, MultiValueVariableState, VariableGetOptionsArgs } from './MultiValueVariable';
+
+export interface DataSourceVariableState extends MultiValueVariableState {
+  query: string;
+  regex: string;
+  refresh: VariableRefresh;
```

Comment:
don't think we need this option, this type of variable should always refresh on dashboard load

Reply:
removed!